runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) 2014 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.xwalk.core;
+
+import android.content.Context;
+import android.content.pm.PackageManager.NameNotFoundException;
+import android.util.Log;
+
+import junit.framework.Assert;
+
+import org.xwalk.core.XWalkLibraryListener.LibraryStatus;
+
+class XWalkCoreWrapper implements ReflectExceptionHandler {
+    public static final String XWALK_APK_PACKAGE = "org.xwalk.core";
+    public static final String WRAPPER_PACKAGE = "org.xwalk.core";
+    public static final String BRIDGE_PACKAGE = "org.xwalk.core.internal";
+
+    private static XWalkCoreWrapper sProvisionalInstance;
+    private static XWalkCoreWrapper sInstance;
+    private static XWalkLibraryListener sListener;
+
+    private int mSdkVersion;
+    private int mMinSdkVersion;
+
+    private Object mCore;
+    private LibraryStatus mCoreStatus;
+    private ClassLoader mBridgeLoader;
+    private Context mBridgeContext;
+
+    public static XWalkCoreWrapper getInstance() {
+        return sInstance;
+    }
+
+    public static XWalkCoreWrapper getProvisionalInstance() {
+        return sProvisionalInstance;
+    }
+
+    public static boolean checkEmbeddedMode() {
+        Assert.assertNull(sInstance);
+        sProvisionalInstance = new XWalkCoreWrapper(-1);
+        return sProvisionalInstance.findEmbeddedCore();
+    }
+
+    public static void check() {
+        check(-1);
+    }
+
+    public static void check(int minSdkVersion) {
```

Comment:
This is incorrect assumption, public means it can be used from anyplace, http://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html. We should avoid this api use outside the core_internal package.